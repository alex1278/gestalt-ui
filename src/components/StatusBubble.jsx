import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import FontIcon from 'react-md/lib/FontIcons';

const ContainerStatusStyle = styled.div`
  text-align: left;
  min-width: 115px;
  max-width: 130px;
  border-radius: 20px;
  padding: 2px;
  color: white;
  font-weight: 600;
  font-size: 12px;
  text-transform: uppercase;
  ${props => props.status === 'RUNNING' && `background-color: ${props.theme.colors['$md-blue-grey-400']}`};
  ${props => props.status === 'HEALTHY' && `background-color: ${props.theme.colors['$md-green-a700']}`};
  ${props => props.status === 'UNHEALTHY' && `background-color: ${props.theme.colors['$md-red-a400']}`};
  ${props => props.status === 'SCALING' && `background-color: ${props.theme.colors['$md-blue-500']}`};
  ${props => props.status === 'SUSPENDED' && `background-color: ${props.theme.colors['$md-orange-500']}`};
  ${props => props.status === 'LOST' && `background-color: ${props.theme.colors['$md-red-a400']}`};
  ${props => props.status === 'ERROR' && `background-color: ${props.theme.colors['$md-red-a400']}`};
  ${props => props.status === 'ACTIVE' && `background-color: ${props.theme.colors['$md-blue-grey-400']}`};
  ${props => props.status === 'FAILED' && `background-color: ${props.theme.colors['$md-red-a400']}`};
  ${props => props.status === 'PENDING' && `background-color: ${props.theme.colors['$md-orange-500']}`};
  ${props => props.status === 'DELETED' && `background-color: ${props.theme.colors['$md-grey-900']}`};
  ${props => props.status === 'DISABLED' && `background-color: ${props.theme.colors['$md-grey-500']}`};
  ${props => props.status === 'LAUNCHED' && `background-color: ${props.theme.colors['$md-grey-300']}`};
  ${props => props.status === 'UNKNOWN' && `background-color: ${props.theme.colors['$md-red-a400']}`};

  i {
    color: white;
    margin-left: 1px;
  }

  span {
    text-align: center;
    line-height: 20px;
    margin-left: 8px;

    @media (min-width: 0) and (max-width: 1025px) {
      line-height: 24px;
    }
  }
`;

const ContainerStatus = (props) => {
  const setIcon = () => {
    switch (props.status.toUpperCase()) {
      case 'RUNNING':
        return 'check_circle';
      case 'HEALTHY':
        return 'check_circle';
      case 'UNHEALTHY':
        return 'new_releases';
      case 'SCALING':
        return 'access_time';
      case 'SUSPENDED':
        return 'info';
      case 'LOST':
        return 'new_releases';
      case 'ERROR':
        return 'new_releases';
      case 'ACTIVE':
        return 'check_circle';
      case 'FAILED':
        return 'new_releases';
      case 'PENDING':
        return 'access_time';
      case 'DELETED':
        return 'clear';
      case 'DISABLED':
        return 'remove_circle';
      case 'UNKNOWN':
        return 'help_outline';
      default:
        return 'info';
    }
  };

  return (
    <ContainerStatusStyle status={props.status.toUpperCase()}>
      <FontIcon>{setIcon()}</FontIcon><span>{props.status}</span>
    </ContainerStatusStyle>
  );
};

ContainerStatus.propTypes = {
  status: PropTypes.string,
};

ContainerStatus.defaultProps = {
  status: '',
};

export default ContainerStatus;
