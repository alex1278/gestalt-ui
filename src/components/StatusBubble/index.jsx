import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import FontIcon from 'react-md/lib/FontIcons';

const ContainerStatusStyle = styled.div`
  text-align: left;
  min-width: 110px;
  max-width: 130px;
  border-radius: 20px;
  padding: 2px;
  color: white;
  font-weight: 600;
  font-size: 12px;
  ${props => props.status === 'RUNNING' && `background-color: ${props.theme.colors['$md-green-a700']}`};
  ${props => props.status === 'LAUNCHED' && `background-color: ${props.theme.colors['$md-blue-500']}`};
  ${props => props.status === 'HEALTHY' && `background-color: ${props.theme.colors['$md-green-a700']}`};
  ${props => props.status === 'UNHEALTHY' && `background-color: ${props.theme.colors['$md-orange-a400']}`};
  ${props => props.status === 'SCALING' && `background-color: ${props.theme.colors['$md-blue-500']}`};
  ${props => props.status === 'SUSPENDED' && `background-color: ${props.theme.colors['$md-orange-a400']}`};
  ${props => props.status === 'LOST' && `background-color: ${props.theme.colors['$md-red-a400']}`};
  ${props => props.status === 'ERROR' && `background-color: ${props.theme.colors['$md-red-a400']}`};
  ${props => props.status === 'Active' && `background-color: ${props.theme.colors['$md-green-a700']}`};
  ${props => props.status === 'Failed' && `background-color: ${props.theme.colors['$md-red-a400']}`};

  i {
    color: white;
  }

  span {
    text-align: center;
    line-height: 20px;
    margin-left: 8px;
  }
`;

const ContainerStatus = (props) => {
  const setIcon = () => {
    switch (props.status) {
      case 'RUNNING':
        return 'check_circle';
      case 'LAUNCHED':
        return 'info';
      case 'HEALTHY':
        return 'check_circle';
      case 'UNHEALTHY':
        return 'error_outline';
      case 'SCALING':
        return 'access_time';
      case 'SUSPENDED':
        return 'info';
      case 'LOST':
        return 'new_releases';
      case 'ERROR':
        return 'new_releases';
      case 'Active':
        return 'check_circle';
      case 'Failed':
        return 'new_releases';
      default:
        return 'info';
    }
  };

  return (
    <ContainerStatusStyle status={props.status}>
      <FontIcon>{setIcon()}</FontIcon><span>{props.status}</span>
    </ContainerStatusStyle>
  );
};

ContainerStatus.propTypes = {
  status: PropTypes.string,
};

ContainerStatus.defaultProps = {
  status: '',
};

export default ContainerStatus;
