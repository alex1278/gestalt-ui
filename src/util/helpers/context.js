/**
 * generateContextEntityState
 * @param {*} params
 */
export function generateContextEntityState(params) {
  const entity = {
    id: params.environmentId || params.workspaceId || null,
    key: null,
  };

  if (params.workspaceId && params.environmentId) {
    entity.key = 'environments';
  }

  if (params.workspaceId && !params.environmentId) {
    entity.key = 'workspaces';
  }

  // if (params.workspaceId && params.environmentId && params.providerId) {
  //   entity.id = params.providerId;
  //   entity.key = 'providers';
  // }

  return entity;
}
