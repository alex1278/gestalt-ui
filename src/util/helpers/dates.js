import moment from 'moment';

export const getNow = () => moment();

export const getAgo = (num = 0, unit = 'day') =>
  moment().subtract(num, unit);

export const isRecentWithin = (timestamp, num, unit) =>
  moment(timestamp).isBetween(getAgo(num, unit), getNow());
