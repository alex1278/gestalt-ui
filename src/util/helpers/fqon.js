// sad hack for parsing url

export default function fqonParent(organization) {
  if (organization.org) {
    const urlParser = document.createElement('a');
    urlParser.href = organization.org.href;
    return urlParser.pathname.replace('/', '').split('/')[0];
  }

  return organization;
}
