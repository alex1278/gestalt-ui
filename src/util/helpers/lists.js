export function insertItem(array, payload) {
  return [...array, payload];
}

export function removeItem(array, payload) {
  const newArray = array.slice();
  newArray.splice(newArray.findIndex(a => a === payload), 1);

  return newArray;
}

export function removeItemById(array, id, keyField = 'id') {
  const newArray = array.slice();
  newArray.splice(newArray.findIndex(a => a[keyField] === id), 1);

  return newArray;
}

export function updateItem(array, payload) {
  return array.map((item) => {
    if (item.id === payload.id) {
      return {
        ...payload,
      };
    }

    // Leave every other item unchanged
    return item;
  });
}
