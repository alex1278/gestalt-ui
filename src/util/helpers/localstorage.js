/**
 * getItem
 * @param {*} key
 */
export const getItem = (key) => {
  try {
    return localStorage.getItem(key);
  } catch (err) {
    return undefined;
  }
};

/**
 * saveItem
 * @param {*} key
 * @param {*} value
 */
export const saveItem = (key, value) => {
  try {
    localStorage.setItem(key, value);
  } catch (err) {
    // ignore write errors
  }

  return undefined;
};

/**
 * removeItem
 * @param {*} key
 * @param {*} value
 */
export const removeItem = (key) => {
  try {
    localStorage.removeItem(key);
  } catch (err) {
    // ignore write errors
  }

  return undefined;
};
