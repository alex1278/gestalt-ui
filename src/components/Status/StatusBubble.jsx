import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { media } from 'util/helpers/media';
import statusMap from './statusMap';

const ContainerStatusStyle = styled.div`
  display: flex;
  align-items: center;
  min-width: 115px;
  max-width: 130px;
  font-weight: 400;
  font-size: 11px;
  text-transform: uppercase;
  color: ${props => props.theme.colors[props.color]};

  svg {
    fill: ${props => props.theme.colors[props.color]};
  }

  div {
    margin-left: 8px;
    height: 14px;
    ${() => media.xs`
      height: 18px;
    `};
    ${() => media.sm`
      height: 18px;
    `};
  }
`;

const StatusDetail = styled.div`
  display: block;
  padding-top: 6px;
`;

const ContainerStatus = ({ status, statusDetail }) => {
  const style = statusMap(status);

  return (
    <React.Fragment>
      <ContainerStatusStyle color={style.color}>
        {style.icon}
        <div>{status}</div>
      </ContainerStatusStyle>
      {statusDetail.reason && <StatusDetail>{`${statusDetail.stateId}-${statusDetail.reason}`}</StatusDetail>}
    </React.Fragment>
  );
};

ContainerStatus.propTypes = {
  status: PropTypes.string,
  statusDetail: PropTypes.object,
};

ContainerStatus.defaultProps = {
  status: 'PENDING',
  statusDetail: {},
};

export default ContainerStatus;
