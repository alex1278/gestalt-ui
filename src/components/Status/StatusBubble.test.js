import React from 'react';
import StatusBubble from './StatusBubble';
import { renderWithTheme } from '../../../test/helpers';

describe('Components::StatusBubble::StatusBubble', () => {
  it('mounts with default props', () => {
    const { container } = renderWithTheme(<StatusBubble />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly when a status is passed', () => {
    const { container } = renderWithTheme(<StatusBubble status="running" />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
