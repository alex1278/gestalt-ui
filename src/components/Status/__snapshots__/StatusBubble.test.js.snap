// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components::StatusBubble::StatusBubble mounts with default props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 115px;
  max-width: 130px;
  font-weight: 400;
  font-size: 11px;
  text-transform: uppercase;
  color: #ff9800;
}

.c0 svg {
  fill: #ff9800;
}

.c0 div {
  margin-left: 8px;
  height: 14px;
}

@media (min-width:0) and (max-width:599px) {
  .c0 div {
    height: 18px;
  }
}

@media (min-width:600px) and (max-width:959px) {
  .c0 div {
    height: 18px;
  }
}

<div
  class="c0"
  color="$md-orange-500"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root-1 MuiSvgIcon-fontSizeSmall-8"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
    />
  </svg>
  <div>
    PENDING
  </div>
</div>
`;

exports[`Components::StatusBubble::StatusBubble renders correctly when a status is passed 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 115px;
  max-width: 130px;
  font-weight: 400;
  font-size: 11px;
  text-transform: uppercase;
  color: #00c853;
}

.c0 svg {
  fill: #00c853;
}

.c0 div {
  margin-left: 8px;
  height: 14px;
}

@media (min-width:0) and (max-width:599px) {
  .c0 div {
    height: 18px;
  }
}

@media (min-width:600px) and (max-width:959px) {
  .c0 div {
    height: 18px;
  }
}

<div
  class="c0"
  color="$md-green-a700"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root-1 MuiSvgIcon-fontSizeSmall-8"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M0 0h24v24H0V0zm0 0h24v24H0V0z"
      fill="none"
    />
    <path
      d="M16.59 7.58L10 14.17l-3.59-3.58L5 12l5 5 8-8zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
    />
  </svg>
  <div>
    running
  </div>
</div>
`;
